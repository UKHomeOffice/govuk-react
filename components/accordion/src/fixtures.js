import React, { useMemo, useReducer } from 'react';
import PropTypes from 'prop-types';
import Link from '@govuk-react/link';
import ListNavigation from '@govuk-react/list-navigation';
import Paragraph from '@govuk-react/paragraph';

import Accordion from '.';

const sharedPropTypes = {
  initialState: PropTypes.shape({}).isRequired,
};

const constructFull = ({ bool, state }) =>
  Object.keys(state).reduce(
    (prev, key) => ({
      ...prev,
      [key]: bool,
    }),
    {}
  );

const reducer = (state, newState) => ({ ...state, ...newState });

const useLegacyState = state => useReducer(reducer, state);

const checkOpen = state => Object.values(state).every(Boolean);

const useAccordion = initialState => {
  const [state, setState] = useLegacyState(initialState);

  const objOpen = useMemo(() => constructFull({ bool: true, state }), [initialState, state.length]);
  const objClose = useMemo(() => constructFull({ bool: false, state }), [initialState, state.length]);

  const areAllOpen = useMemo(() => checkOpen(state), [state]);

  const toggleAll = e => {
    e.preventDefault();
    return setState(!areAllOpen ? objOpen : objClose);
  };

  const individualAccordionSetState = property => () =>
    setState({
      [property]: !state[property],
    });

  return {
    areAllOpen,
    individualAccordionSetState,
    toggleAll,
    state,
  };
};

const ExampleWithoutSummaries = ({ initialState }) => {
  const { areAllOpen, individualAccordionSetState, toggleAll, state } = useAccordion(initialState);

  return (
    <Accordion>
      <Accordion.OpenAll href="#" onClick={toggleAll}>
        {`${areAllOpen ? 'Close' : 'Open'} all`}
      </Accordion.OpenAll>
      <Accordion.Group
        heading="Writing well for the web"
        expanded={state.accordionOne}
        onClick={individualAccordionSetState('accordionOne')}
      >
        <Paragraph mb={0}>This is the content for writing well for the web.</Paragraph>
      </Accordion.Group>
      <Accordion.Group
        heading="Writing well for specialists"
        expanded={state.accordionTwo}
        onClick={individualAccordionSetState('accordionTwo')}
      >
        <Paragraph mb={0}>This is the content for writing well for the specialists.</Paragraph>
      </Accordion.Group>
      <Accordion.Group
        heading="Know your audience"
        expanded={state.accordionThree}
        onClick={individualAccordionSetState('accordionThree')}
      >
        <Paragraph mb={0}>This is the content for Know you audience.</Paragraph>
      </Accordion.Group>
      <Accordion.Group
        heading="How people read"
        expanded={state.accordionFour}
        onClick={individualAccordionSetState('accordionFour')}
      >
        <Paragraph mb={0}>This is the content for How people read.</Paragraph>
      </Accordion.Group>
    </Accordion>
  );
};

ExampleWithoutSummaries.propTypes = sharedPropTypes;

const ExampleWithSummaries = ({ initialState }) => {
  const { areAllOpen, individualAccordionSetState, toggleAll, state } = useAccordion(initialState);

  return (
    <Accordion>
      <Accordion.OpenAll href="#" onClick={toggleAll}>
        {`${areAllOpen ? 'Close' : 'Open'} all`}
      </Accordion.OpenAll>
      <Accordion.Group
        heading="Understanding agile project management"
        summary="Introductions, methods, core features."
        expanded={state.accordionOne}
        onClick={individualAccordionSetState('accordionOne')}
      >
        <ListNavigation listStyleType="none">
          {[
            'Agile and government services: an introduction',
            'Agile methods: an introduction',
            'Core principles of agile',
          ].map(elem => (
            <Link key={elem} href={`#myprefix-${elem}`}>
              {elem}
            </Link>
          ))}
        </ListNavigation>
      </Accordion.Group>
      <Accordion.Group
        heading="Working with agile methods"
        summary="Workspaces, tools and techniques, user stories, planning."
        expanded={state.accordionTwo}
        onClick={individualAccordionSetState('accordionTwo')}
      >
        <ListNavigation listStyleType="none">
          {[
            'Creating an agile working environment',
            'Agile tools and techniques',
            'Set up a team wall',
            'Writing user stories',
            'Planning in agile',
            'Deciding on priorities',
            'Developing a roadmap',
          ].map(elem => (
            <Link key={elem} href={`#myprefix-${elem}`}>
              {elem}
            </Link>
          ))}
        </ListNavigation>
      </Accordion.Group>
      <Accordion.Group
        heading="Governing agile services"
        summary="Principles, measuring progress, spending money."
        expanded={state.accordionThree}
        onClick={individualAccordionSetState('accordionThree')}
      >
        <ListNavigation listStyleType="none">
          {[
            'How the discovery phase works',
            'Measuring and reporting progress',
            'Spend controls: check if you need approval to spend money on a service',
            'Spend controls: apply for approval to spend money on a service',
            'Working across organisational boundaries',
          ].map(elem => (
            <Link key={elem} href={`#myprefix-${elem}`}>
              {elem}
            </Link>
          ))}
        </ListNavigation>
      </Accordion.Group>
      <Accordion.Group
        heading="Phases of an agile project"
        summary="Discovery, alpha, beta, live and retirement."
        expanded={state.accordionFour}
        onClick={individualAccordionSetState('accordionFour')}
      >
        <ListNavigation listStyleType="none">
          {[
            'Governance principles for agile service delivery',
            'How the alpha phase works',
            'How the beta phase works',
            'How the live phase works',
            'Retiring your service',
          ].map(elem => (
            /* eslint-disable-next-line jsx-a11y/anchor-is-valid */
            <Link key={elem} href={`#myprefix-${elem}`}>
              {elem}
            </Link>
          ))}
        </ListNavigation>
      </Accordion.Group>
    </Accordion>
  );
};

ExampleWithSummaries.propTypes = sharedPropTypes;

class AccordionClassComponent extends React.Component {
  constructor() {
    super();
    this.state = {
      accordionOne: false,
      accordionTwo: false,
      accordionThree: false,
      accordionFour: false,
    };
    this.individualAccordionSetState = this.individualAccordionSetState.bind(this);
    this.toggleAll = this.toggleAll.bind(this);
  }

  toggleAll() {
    return this.setState(state =>
      Object.keys(state).reduce(
        (prev, key) => ({
          ...prev,
          [key]: !this.isAllOpen,
        }),
        {}
      )
    );
  }

  individualAccordionSetState(property) {
    return () =>
      this.setState(state => ({
        [property]: !state[property],
      }));
  }

  render() {
    this.isAllOpen = checkOpen(this.state);
    return (
      <Accordion>
        <Accordion.OpenAll href="#" onClick={this.toggleAll}>
          {`${this.isAllOpen ? 'Close' : 'Open'} all`}
        </Accordion.OpenAll>
        <Accordion.Group
          heading="Writing well for the web"
          expanded={this.state.accordionOne}
          onClick={this.individualAccordionSetState('accordionOne')}
        >
          <Paragraph mb={0}>This is the content for writing well for the web.</Paragraph>
        </Accordion.Group>
        <Accordion.Group
          heading="Writing well for specialists"
          expanded={this.state.accordionTwo}
          onClick={this.individualAccordionSetState('accordionTwo')}
        >
          <Paragraph mb={0}>This is the content for writing well for the specialists.</Paragraph>
        </Accordion.Group>
        <Accordion.Group
          heading="Know your audience"
          expanded={this.state.accordionThree}
          onClick={this.individualAccordionSetState('accordionThree')}
        >
          <Paragraph mb={0}>This is the content for Know you audience.</Paragraph>
        </Accordion.Group>
        <Accordion.Group
          heading="How people read"
          expanded={this.state.accordionFour}
          onClick={this.individualAccordionSetState('accordionFour')}
        >
          <Paragraph mb={0}>This is the content for How people read.</Paragraph>
        </Accordion.Group>
      </Accordion>
    );
  }
}

export { AccordionClassComponent, ExampleWithoutSummaries, ExampleWithSummaries };
