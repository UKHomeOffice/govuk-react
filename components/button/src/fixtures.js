import React from 'react';
import { boolean } from '@storybook/addon-knobs/react';
import { ButtonArrow } from '@govuk-react/icons';
import Button from '.';

export default () => <Button>Save and continue</Button>;

export const ButtonStart = () => <Button start={boolean('Start', true)}>Save and continue</Button>;

export const ButtonStartIcon = () => (
  <Button icon={<ButtonArrow />} start={boolean('Start', true)}>
    Start now
  </Button>
);

export const ButtonDisabled = () => <Button disabled={boolean('Disabled', true)}>Disabled primary button</Button>;

export const ButtonDisabledStartIcon = () => (
  <Button
    disabled={boolean('Disabled', true)}
    start={boolean('Start', true)}
    icon={<ButtonArrow />}
  >
    Start now
  </Button>
);
