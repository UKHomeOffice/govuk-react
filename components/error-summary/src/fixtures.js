import React from 'react';
import { text } from '@storybook/addon-knobs/react';
import styled from 'react-emotion';

import ErrorSummary from '.';

export const heading = 'Message to alert the user to a problem goes here';
export const description = 'Optional description of the errors and how to correct them';
export const errors = [
  'Descriptive link to the target question with an error',
  'Descriptive link to the other target question with an error',
];

const StyledDiv = styled('div')({
  marginBottom: '150px',
});

export default class ErrorSummaryExample extends React.Component {
  errors = [
    {
      id: 0,
      handleScrollToElement: () => this.targetQuestion.scrollIntoView(),
      text: 'Descriptive link to the target question with an error',
    },
    {
      id: 1,
      handleScrollToElement: () => this.otherTargetQuestion.scrollIntoView(),
      text: 'Descriptive link to the other target question with an error',
    },
  ];

  render() {
    return (
      <div>
        <ErrorSummary
          heading={text('heading', heading)}
          description={text('description', description)}
          errors={this.errors}
        />
        <StyledDiv innerRef={(node) => { this.targetQuestion = node; }}>
          Target Question
        </StyledDiv>
        <div ref={(node) => { this.otherTargetQuestion = node; }}>
          Other Target Question
        </div>
      </div>
    );
  }
}
