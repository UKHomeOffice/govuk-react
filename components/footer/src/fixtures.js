import React from 'react';
import { boolean, text } from '@storybook/addon-knobs/react';

import { BrowserRouter, Link, Route } from 'react-router-dom';

import Footer from '.';

const withRoutes = component => (
  <BrowserRouter>
    {component}
    <Route path="/footer-nav-item-2" render={() => 'Footer Navigation item 2'} />
    <Route path="/footer-meta-item-2" render={() => 'Footer Meta item 2'} />
  </BrowserRouter>
);

export default () => <Footer />;

const metaLinks = (
  <Footer.MetaLinks heading="Support links">
    <Footer.Link href="/">Item 1</Footer.Link>
    <Footer.Link to="/footer-meta-item-2" as={Link}>
      Item 2 (Router Link)
    </Footer.Link>
    <Footer.Link href="/">Item 3</Footer.Link>
  </Footer.MetaLinks>
);

const metaCustom = (
  <Footer.MetaCustom>
    Built by the <Footer.Link href="/">Government Digital Service</Footer.Link>
  </Footer.MetaCustom>
);

const metaLinksAndCustom = (
  <React.Fragment>
    {metaLinks}
    {metaCustom}
  </React.Fragment>
);

const navigation = (
  <Footer.Navigation>
    <Footer.NavigationLinks heading="Two column list" listColumns="2">
      <Footer.Link href="/">Navigation item 1</Footer.Link>
      <Footer.Link to="/footer-nav-item-2" as={Link}>
        Navigation item 2 (Router Link)
      </Footer.Link>
      <Footer.Link href="/">Navigation item 3</Footer.Link>
      <Footer.Link href="/">Navigation item 4</Footer.Link>
      <Footer.Link href="/">Navigation item 5</Footer.Link>
      <Footer.Link href="/">Navigation item 6</Footer.Link>
    </Footer.NavigationLinks>
    <Footer.NavigationLinks heading="Single column list">
      <Footer.Link href="/">Navigation item 1</Footer.Link>
      <Footer.Link href="/">Navigation item 2</Footer.Link>
      <Footer.Link href="/">Navigation item 3</Footer.Link>
    </Footer.NavigationLinks>
  </Footer.Navigation>
);

export const FooterWithMetaLinks = () => withRoutes(<Footer meta={metaLinks} />);

export const FooterWithMetaCustom = () => <Footer meta={metaCustom} />;

export const FooterWithMetaLinksAndCustom = () => withRoutes(<Footer meta={metaLinksAndCustom} />);

export const FooterWithNavigation = () => withRoutes(<Footer>{navigation}</Footer>);

export const FooterWithNavigationAndMeta = () => withRoutes(<Footer meta={metaLinksAndCustom}>{navigation}</Footer>);
