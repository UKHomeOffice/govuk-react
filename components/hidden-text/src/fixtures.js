import React from 'react';
import { text } from '@storybook/addon-knobs/react';
import Paragraph from '@govuk-react/paragraph';

import HiddenText from '.';

export const summaryText = 'Help with nationality';

export const paragraphs = [
  'If you’re not sure about your nationality, try to find out from an official document like a passport or national ID card.',
  'We need to know your nationality so we can work out which elections you’re entitled to vote in. If you can’t provide your nationality, you’ll have to send copies of identity documents through the post.',
];

export default () => (
  <HiddenText summaryText={summaryText}>
    { paragraphs.map((paragraphText, index) =>
      (index === paragraphs.length - 1
       ? <Paragraph key={`paragraphText${index + 1}`} mb={0}>{text(`paragraphText${index + 1}`, paragraphText)}</Paragraph>
       : <Paragraph key={`paragraphText${index + 1}`}>{text(`paragraphText${index + 1}`, paragraphText)}</Paragraph>
      ))}
  </HiddenText>
);
