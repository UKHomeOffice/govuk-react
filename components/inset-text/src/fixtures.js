import React from 'react';
import { boolean } from '@storybook/addon-knobs/react';
import InsetText from '.';

const insetText = 'It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application.';

export default () => <InsetText>{insetText}</InsetText>;

export const InsetTextNarrowBorder = () => <InsetText isNarrow={boolean('IsNarrow', true)}>{insetText}</InsetText>;
