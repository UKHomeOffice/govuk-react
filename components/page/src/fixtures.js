import React from 'react';
import { BackLink, H1 } from 'govuk-react';
import Page from '.';

const MyPage = () => (
  <Page
    beforeChildren={<BackLink href="#" />}
  >
    <H1>Page Title</H1>
  </Page>);

export default MyPage;
