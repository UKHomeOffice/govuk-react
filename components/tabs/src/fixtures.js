import React, { Component } from 'react';
import { H2 } from '@govuk-react/header';
import Table from '@govuk-react/table';

import Tabs from '.';

const { Tab, TabList, TabPanel, TabsContainer, TabsTitle } = Tabs;

const arrTitles = ['Past day', 'Past week', 'Past month', 'Past year'];

const flip2dArray = (prev, next) =>
  next.map((item, i) => [...(prev[i] || []), next[i]]);

const tableHead = (
  <Table.Row>
    <Table.CellHeader>Case manager</Table.CellHeader>
    <Table.CellHeader>Cases opened</Table.CellHeader>
    <Table.CellHeader>Cases closed</Table.CellHeader>
  </Table.Row>
);

class TableTabs extends Component {
  state = {
    tabIndex: 0,
  }


  setTabIndex = newTabIndex => this.setState({
    tabIndex: newTabIndex,
  })

  render() {
    const { tabIndex } = this.state;
    return (
      <TabsContainer>
        <TabsTitle>Content</TabsTitle>
        <TabList>
          {arrTitles.map((label, index) => (
            <Tab
              key={`${label}-tabHeader`}
              onClick={() => this.setTabIndex(index)}
              isActive={tabIndex === index}
            >
              {label}
            </Tab>
          ))}
        </TabList>
        {[
          {
            title: 'Past day',
            arr: [[3, 1, 2], [0, 0, 0]],
            id: 'past-day',
          },
          {
            title: 'Past week',
            arr: [[24, 16, 24], [18, 20, 27]],
            id: 'past-week',
          },
          {
            title: 'Past month',
            arr: [[98, 122, 126], [95, 131, 142]],
            id: 'past-month',
          },
          {
            title: 'Past year',
            arr: [[1380, 1129, 1539], [1472, 1083, 1265]],
            id: 'past-year',
          },
        ].map(({ arr, id, title }, index) => (
          <TabPanel isActive={tabIndex === index} key={`${title}-tabPanel`} id={id}>
            <H2>{title}</H2>
            <Table
              head={tableHead}
            >
              {[['David Francis', 'Paul Farmer', 'Rita Patel'], ...arr]
                .reduce(flip2dArray, [])
                .map(innerArr => (
                  <Table.Row key={`${innerArr.join()}-col`}>
                    {innerArr.map(elem => (
                      <Table.Cell key={`${elem}-row`}>{elem}</Table.Cell>
                    ))}
                  </Table.Row>
                ))}
            </Table>
          </TabPanel>
        ))}
      </TabsContainer>
    );
  }
}

class SimpleTabs extends Component {
  constructor() {
    super();
    this.state = { tabIndex: 0 };
    this.setTabIndex = this.setTabIndex.bind(this);
  }

  setTabIndex(newTabIndex) {
    this.setState({
      tabIndex: newTabIndex,
    });
  }

  render() {
    const { tabIndex } = this.state;
    return (
      <TabsContainer>
        <TabList>
          <Tab
            onClick={() => this.setTabIndex(0)}
            isActive={tabIndex === 0}
            href="#first-panel"
          >
            Title 1
          </Tab>
          <Tab
            onClick={() => this.setTabIndex(1)}
            isActive={tabIndex === 1}
            href="#second-panel"
          >
            Title 2
          </Tab>
        </TabList>
        <TabPanel isActive={tabIndex === 0} id="first-panel">TabPanel content 1</TabPanel>
        <TabPanel isActive={tabIndex === 1} id="second-panel" >TabPanel content 2</TabPanel>
      </TabsContainer>
    );
  }
}

class SimpleMapTabs extends Component {
  constructor() {
    super();
    this.state = { tabIndex: 0 };
    this.setTabIndex = this.setTabIndex.bind(this);
  }

  setTabIndex(newTabIndex) {
    this.setState({
      tabIndex: newTabIndex,
    });
  }

  render() {
    const { tabIndex } = this.state;
    return (
      <TabsContainer>
        <TabList>
          {[
            {
              content: 'Title 1',
              href: '#first-panel'
            },
            {
              content: 'Title 2',
              href: '#second-panel'
            },
          ].map(({ content, href }, index) => (
              <Tab
                onClick={() => this.setTabIndex(index)}
                isActive={tabIndex === index}
                href={href}
              >
                {content}
              </Tab>
            ))
          }
        </TabList>
        {[
            {
              content: 'TabPanel content 1',
              id: 'first-panel'
            },
            {
              content: 'TabPanel content 2',
              id: 'second-panel'
            },
          ].map(({ content, id }, index) => (
              <TabPanel
                isActive={tabIndex === index}
                id={id}
              >
                {content}
              </TabPanel>
            ))
          }
        </TabsContainer>
    );
  }
}

class ProposedClassPropertiesPlugin extends Component {
  state = { tabIndex: 0 };
  
  setTabIndex = newTabIndex => this.setState({
    tabIndex: newTabIndex,
  });

  render() {
    const { tabIndex } = this.state;
    return (
      <Tabs>
        <TabList>
          {[
            {
              content: 'Title 1',
              href: '#first-panel'
            },
            {
              content: 'Title 2',
              href: '#second-panel'
            },
          ].map(({ content, href }, index) => (
              <Tab
                onClick={() => this.setTabIndex(index)}
                isActive={tabIndex === index}
                href={href}
              >
                {content}
              </Tab>
            ))
          }
        </TabList>
        {[
            {
              content: 'TabPanel content 1',
              id: 'first-panel'
            },
            {
              content: 'TabPanel content 2',
              id: 'second-panel'
            },
          ].map(({ content, id }, index) => (
              <TabPanel
                isActive={tabIndex === index}
                id={id}
              >
                {content}
              </TabPanel>
            ))
          }
        </Tabs>
    );
  }
}

export { ProposedClassPropertiesPlugin, SimpleTabs, SimpleMapTabs, TableTabs };
