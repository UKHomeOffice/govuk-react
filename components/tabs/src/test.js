import React from 'react';
import { cleanup, render } from 'react-testing-library';

import { TableTabs } from './fixtures';

afterEach(cleanup);

describe('Tabs', () => {
  it('matches wrapper snapshot', () => {
    const {
      asFragment,
    } = render(<TableTabs />);
    expect(asFragment()).toMatchSnapshot();
  });

  // TODO: test for clicking between tabs above media Queries width.
  // Not currently possible with jsdom
});
