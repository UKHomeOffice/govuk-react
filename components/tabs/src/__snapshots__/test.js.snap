// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs matches wrapper snapshot 1`] = `
<DocumentFragment>
  .c7 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 24px;
  line-height: 1.0416666666666667;
  display: block;
  margin-top: 0;
  margin-bottom: 20px;
}

.c9 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
  font-weight: 700;
}

.c9:last-child {
  padding-right: 0;
}

.c10 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
}

.c10:last-child {
  padding-right: 0;
}

.c8 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  margin-bottom: 20px;
}

.c3 {
  margin-left: 25px;
}

.c3:before {
  content: 'â€” ';
  margin-left: -5;
  padding-right: 5px;
}

.c4 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: inline-block;
  padding-top: 10px;
  padding-bottom: 10px;
  color: #0b0c0c;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
  background-color: #ffbf47;
}

.c4:link {
  color: #005ea5;
}

.c4:visited {
  color: #4c2c92;
}

.c4:hover {
  color: #2b8cc4;
}

.c4:active {
  color: #2b8cc4;
}

.c4:focus {
  color: #0b0c0c;
}

.c5 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: inline-block;
  padding-top: 10px;
  padding-bottom: 10px;
}

.c5:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
  background-color: #ffbf47;
}

.c5:link {
  color: #005ea5;
}

.c5:visited {
  color: #4c2c92;
}

.c5:hover {
  color: #2b8cc4;
}

.c5:active {
  color: #2b8cc4;
}

.c5:focus {
  color: #0b0c0c;
}

.c2 {
  margin: 0;
  padding: 0;
  list-style: none;
}

.c2 marginBottom {
  margin-bottom: 20px;
}

.c6 {
  display: block;
}

.c6 marginBottom {
  mobile-bottom: 30px;
}

.c11 {
  display: block;
}

.c11 marginBottom {
  mobile-bottom: 30px;
}

.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
}

.c0 marginTop {
  margin-top: 5px;
}

.c0 marginBottom {
  margin-bottom: 20px;
}

.c1 {
  font-size: 19px;
  margin-bottom: 5px;
}

@media print {
  .c7 {
    color: #000;
  }
}

@media print {
  .c7 {
    font-size: 24px;
    line-height: 1.05;
  }
}

@media only screen and (min-width:641px) {
  .c7 {
    font-size: 36px;
    line-height: 1.1111111111111112;
  }
}

@media only screen and (min-width:641px) {
  .c7 {
    margin-bottom: 30px;
  }
}

@media print {
  .c8 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c8 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c8 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c8 {
    margin-bottom: 30px;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    margin-left: 0;
  }

  .c3:before {
    content: none;
  }
}

@media print {
  .c4 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c4 {
    font-family: sans-serif;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    margin-right: 5px;
    float: left;
    color: #0b0c0c;
    text-align: center;
    -webkit-text-decoration: none;
    text-decoration: none;
    margin-top: -5;
    margin-bottom: -1px;
    padding-top: 14px;
    padding-right: 19px;
    padding-bottom: 16px;
    padding-left: 19px;
    border: 1px solid #bfc1c3;
    border-bottom: 0;
    background-color: #ffffff;
  }

  .c4:link {
    color: #0b0c0c;
  }

  .c4:focus {
    background-color: transparent;
  }
}

@media print {
  .c5 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c5 {
    font-family: sans-serif;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    margin-right: 5px;
    float: left;
    color: #0b0c0c;
    text-align: center;
    -webkit-text-decoration: none;
    text-decoration: none;
    padding-right: 20px;
    padding-left: 20px;
    background-color: #f8f8f8;
  }

  .c5:link {
    color: #0b0c0c;
  }

  .c5:focus {
    background-color: #f8f8f8;
  }
}

@media only screen and (min-width:641px) {
  .c2 marginBottom {
    margin-bottom: 30px;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    border-bottom: 1px solid #bfc1c3;
  }

  .c2::after {
    content: '';
    display: block;
    clear: both;
  }
}

@media only screen and (min-width:641px) {
  .c6 marginBottom {
    mobile-bottom: 50px;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    margin-bottom: 0;
    padding-top: 30px;
    padding-right: 20px;
    padding-bottom: 30px;
    padding-left: 20px;
    border: 1px solid #bfc1c3;
    border-top: 0;
  }

  .c6 >:last-child {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c11 marginBottom {
    mobile-bottom: 50px;
  }
}

@media only screen and (min-width:641px) {
  .c11 {
    margin-bottom: 0;
    padding-top: 30px;
    padding-right: 20px;
    padding-bottom: 30px;
    padding-left: 20px;
    border: 1px solid #bfc1c3;
    border-top: 0;
  }

  .c11 >:last-child {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c11 {
    display: none;
  }
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 marginTop {
    margin-top: 5px;
  }
}

@media only screen and (min-width:641px) {
  .c0 marginBottom {
    margin-bottom: 30px;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    display: none;
  }
}

<div
    class="c0"
  >
    <h2
      class="c1"
    >
      Content
    </h2>
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        <a
          class="c4"
          href="#"
        >
          Past day
        </a>
      </li>
      <li
        class="c3"
      >
        <a
          class="c5"
          href="#"
        >
          Past week
        </a>
      </li>
      <li
        class="c3"
      >
        <a
          class="c5"
          href="#"
        >
          Past month
        </a>
      </li>
      <li
        class="c3"
      >
        <a
          class="c5"
          href="#"
        >
          Past year
        </a>
      </li>
    </ul>
    <section
      class="c6"
      id="past-day"
    >
      <h2
        class="c7"
      >
        Past day
      </h2>
      <table
        class="c8"
      >
        <thead
          class=""
        >
          <tr
            class=""
          >
            <th
              class="c9"
            >
              Case manager
            </th>
            <th
              class="c9"
            >
              Cases opened
            </th>
            <th
              class="c9"
            >
              Cases closed
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class=""
          >
            <td
              class="c10"
            >
              David Francis
            </td>
            <td
              class="c10"
            >
              3
            </td>
            <td
              class="c10"
            >
              0
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="c10"
            >
              Paul Farmer
            </td>
            <td
              class="c10"
            >
              1
            </td>
            <td
              class="c10"
            >
              0
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="c10"
            >
              Rita Patel
            </td>
            <td
              class="c10"
            >
              2
            </td>
            <td
              class="c10"
            >
              0
            </td>
          </tr>
        </tbody>
      </table>
    </section>
    <section
      class="c11"
      id="past-week"
    >
      <h2
        class="c7"
      >
        Past week
      </h2>
      <table
        class="c8"
      >
        <thead
          class=""
        >
          <tr
            class=""
          >
            <th
              class="c9"
            >
              Case manager
            </th>
            <th
              class="c9"
            >
              Cases opened
            </th>
            <th
              class="c9"
            >
              Cases closed
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class=""
          >
            <td
              class="c10"
            >
              David Francis
            </td>
            <td
              class="c10"
            >
              24
            </td>
            <td
              class="c10"
            >
              18
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="c10"
            >
              Paul Farmer
            </td>
            <td
              class="c10"
            >
              16
            </td>
            <td
              class="c10"
            >
              20
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="c10"
            >
              Rita Patel
            </td>
            <td
              class="c10"
            >
              24
            </td>
            <td
              class="c10"
            >
              27
            </td>
          </tr>
        </tbody>
      </table>
    </section>
    <section
      class="c11"
      id="past-month"
    >
      <h2
        class="c7"
      >
        Past month
      </h2>
      <table
        class="c8"
      >
        <thead
          class=""
        >
          <tr
            class=""
          >
            <th
              class="c9"
            >
              Case manager
            </th>
            <th
              class="c9"
            >
              Cases opened
            </th>
            <th
              class="c9"
            >
              Cases closed
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class=""
          >
            <td
              class="c10"
            >
              David Francis
            </td>
            <td
              class="c10"
            >
              98
            </td>
            <td
              class="c10"
            >
              95
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="c10"
            >
              Paul Farmer
            </td>
            <td
              class="c10"
            >
              122
            </td>
            <td
              class="c10"
            >
              131
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="c10"
            >
              Rita Patel
            </td>
            <td
              class="c10"
            >
              126
            </td>
            <td
              class="c10"
            >
              142
            </td>
          </tr>
        </tbody>
      </table>
    </section>
    <section
      class="c11"
      id="past-year"
    >
      <h2
        class="c7"
      >
        Past year
      </h2>
      <table
        class="c8"
      >
        <thead
          class=""
        >
          <tr
            class=""
          >
            <th
              class="c9"
            >
              Case manager
            </th>
            <th
              class="c9"
            >
              Cases opened
            </th>
            <th
              class="c9"
            >
              Cases closed
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class=""
          >
            <td
              class="c10"
            >
              David Francis
            </td>
            <td
              class="c10"
            >
              1380
            </td>
            <td
              class="c10"
            >
              1472
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="c10"
            >
              Paul Farmer
            </td>
            <td
              class="c10"
            >
              1129
            </td>
            <td
              class="c10"
            >
              1083
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="c10"
            >
              Rita Patel
            </td>
            <td
              class="c10"
            >
              1539
            </td>
            <td
              class="c10"
            >
              1265
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>
</DocumentFragment>
`;
